### ISSUE TITLE
Policy reporter pods won't start when configured to run with multiple replicas

### ISSUE THREAD

**Issue by an-toine** [fjogeleit/policy-reporter#479]
Hello,

I believe I'm facing a similar issue as the one described in https://github.com/kyverno/policy-reporter/issues/375 : I've deployed kyverno-policy reporter in HA mode, with 3 replicas, and the `policy-reporter` pods are stuck in `CrashLoopBackOff` state.

```
➜  work k get pods -l app.kubernetes.io/name=policy-reporter
NAME                                            READY   STATUS             RESTARTS        AGE
kyverno-policy-reporter-6d5f49848c-7g2m2   0/1     CrashLoopBackOff   41 (5m6s ago)   116m
kyverno-policy-reporter-6d5f49848c-7rcp2   0/1     CrashLoopBackOff   43 (6s ago)     116m
kyverno-policy-reporter-6d5f49848c-vbk46   0/1     CrashLoopBackOff   41 (5m6s ago)   116m
```

Containers are not logging any meaningful error : 

```
➜  work k logs kyverno-policy-reporter-6d5f49848c-7g2m2
{"level":"info","ts":1727450335.447952,"caller":"config/resolver.go:125","msg":"sqlite connection created"}
{"level":"info","ts":1727450335.4713914,"caller":"cmd/run.go:78","msg":"REST api enabled"}
{"level":"info","ts":1727450335.4716072,"caller":"cmd/run.go:83","msg":"metrics enabled"}
{"level":"warn","ts":1727450336.4570658,"caller":"api/handler.go:40","msg":"informers not synced yet, waiting for k8s client to be up"}
{"level":"warn","ts":1727450345.283102,"caller":"api/handler.go:40","msg":"informers not synced yet, waiting for k8s client to be up"}
{"level":"warn","ts":1727450345.2830968,"caller":"api/handler.go:19","msg":"informers not synced yet, waiting for k8s client to complete startup"}
{"level":"warn","ts":1727450355.2842114,"caller":"api/handler.go:40","msg":"informers not synced yet, waiting for k8s client to be up"}
{"level":"warn","ts":1727450355.2842057,"caller":"api/handler.go:19","msg":"informers not synced yet, waiting for k8s client to complete startup"}
{"level":"warn","ts":1727450365.283867,"caller":"api/handler.go:40","msg":"informers not synced yet, waiting for k8s client to be up"}
{"level":"warn","ts":1727450365.2838938,"caller":"api/handler.go:19","msg":"informers not synced yet, waiting for k8s client to complete startup"}
{"level":"warn","ts":1727450365.284794,"caller":"api/handler.go:40","msg":"informers not synced yet, waiting for k8s client to be up"}
```

Kyverno plugin pods are starting just fine : 

```
➜  work k get pods -l app.kubernetes.io/name=kyverno-plugin
NAME                                           READY   STATUS    RESTARTS   AGE
kyverno-kyverno-plugin-56d96858bb-h7gnc   1/1     Running   0          125m
kyverno-kyverno-plugin-56d96858bb-klrbs   1/1     Running   0          125m
kyverno-kyverno-plugin-56d96858bb-p5qf6   1/1     Running   0          125m
```

As soon as I set the replicas count to 1, the policy-reporter pod will start and report a `Running` status.

Versions deployed on the cluster : 
 * Policy reporter chart `v2.24.1`
 * `kyverno/policy-reporter:2.20.1`
 * `kyverno/policy-reporter-kyverno-plugin:1.6.3`
 * `kyverno/kyverno:v1.12.4`
 * Kubernetes 1.29.6

Helm value file : 

```yaml
policy-reporter:
  global:
    namespace: "kyverno"
  image:
    registry: XXXXXX
  metrics:
    enabled: true
  rest:
    enabled: true
  logging:
    encoding: json
  replicaCount: 3
  leaderElection:
    enabled: true
  resources:
    limits:
      memory: 128Mi
    requests:
      memory: 128Mi
      cpu: 50m
  nodeSelector:
    node-role.kubernetes.io/control-plane: "true"
  tolerations:
    - key: "node-role.kubernetes.io/master"
      operator: "Exists"
      effect: "NoSchedule"
  kyvernoPlugin:
    enabled: true
    image:
      registry: XXXXXXX
    replicaCount: 3
    logging:
      encoding: json
    resources:
      limits:
        memory: 64Mi
      requests:
        memory: 64Mi
        cpu: 25m
    nodeSelector:
      node-role.kubernetes.io/control-plane: "true"
    tolerations:
      - key: "node-role.kubernetes.io/master"
        operator: "Exists"
        effect: "NoSchedule"
    blockReports:
      enabled: true
```

I've also tried to increase pods resources and ReadinessProbe `initialDelaySeconds` as suggested here https://kyverno.github.io/policy-reporter/guide/troubleshooting/ but with no luck.

Let me know if you need more details about my deployment.

Antoine

**Comment by fjogeleit** [comment]
Hey, thanks for reporting. I will take a look

**Comment by fjogeleit** [comment]
None of your pods has a log message like?
```
I0928 08:43:00.662090       1 leaderelection.go:254] attempting to acquire leader lease policy-reporter/policy-reporter...
```

And as I see you don't use a push target or the UI?

Can you try removing

```
  leaderElection:
    enabled: true
```

in your setup leader election should not be needed because its mainly responsible to only push notifications with a single pod or updating the DB if an external DB is used.

**Comment by fjogeleit** [comment]
I created a new image tag `2.20.2`. Can you check if this solves your issue?

**Comment by an-toine** [comment]
Once upgraded to image tag `2.20.2`, pods are starting correctly in HA mode : 

```
➜  work k get pods -l app.kubernetes.io/name=policy-reporter
NAME                                            READY   STATUS    RESTARTS   AGE
kyverno-policy-reporter-595b6d685d-7jk8s   1/1     Running   0          9m21s
kyverno-policy-reporter-595b6d685d-gjjxr   1/1     Running   0          9m2s
kyverno-policy-reporter-595b6d685d-jqpxd   1/1     Running   0          9m5s
➜  work k get pod kyverno-policy-reporter-595b6d685d-jqpxd -o yaml | grep -i image:
    image: kyverno/policy-reporter:2.20.2
    image: kyverno/policy-reporter:2.20.2
➜  work k logs -l app.kubernetes.io/name=policy-reporter
{"level":"info","ts":1727686722.8681126,"caller":"config/resolver.go:125","msg":"sqlite connection created"}
{"level":"info","ts":1727686722.8851647,"caller":"cmd/run.go:78","msg":"REST api enabled"}
{"level":"info","ts":1727686722.8854456,"caller":"cmd/run.go:83","msg":"metrics enabled"}
{"level":"info","ts":1727686722.8857431,"caller":"cmd/run.go:153","msg":"start client","worker":5}
{"level":"info","ts":1727686722.9865558,"caller":"kubernetes/policy_report_client.go:63","msg":"informer sync completed"}
{"level":"info","ts":1727686728.4381409,"caller":"config/resolver.go:125","msg":"sqlite connection created"}
{"level":"info","ts":1727686728.4518569,"caller":"cmd/run.go:78","msg":"REST api enabled"}
{"level":"info","ts":1727686728.4523497,"caller":"cmd/run.go:83","msg":"metrics enabled"}
{"level":"info","ts":1727686728.4527817,"caller":"cmd/run.go:153","msg":"start client","worker":5}
{"level":"info","ts":1727686728.5530918,"caller":"kubernetes/policy_report_client.go:63","msg":"informer sync completed"}
{"level":"info","ts":1727686725.5669935,"caller":"config/resolver.go:125","msg":"sqlite connection created"}
{"level":"info","ts":1727686725.5767868,"caller":"cmd/run.go:78","msg":"REST api enabled"}
{"level":"info","ts":1727686725.5771115,"caller":"cmd/run.go:83","msg":"metrics enabled"}
{"level":"info","ts":1727686725.577327,"caller":"cmd/run.go:153","msg":"start client","worker":5}
{"level":"info","ts":1727686725.6783118,"caller":"kubernetes/policy_report_client.go:63","msg":"informer sync completed"}
```

Thanks a lot for your help 🙂 

Antoine

**Comment by fjogeleit** [comment]
Thanks for your feedback, I released v2.24.2 of the Helm Chart with the fixed image.
